xptifw/src/xpti_trace_framework.cpp

```diff
@@ -1179,25 +1238,57 @@ class Tracepoints {
     uint64_t FileId = 0, FuncId = 0;
     int LineNo = 0, ColNo = 0;
 
-    // If the payload's function name is available, add it to the string table
-    // and get its id
-    if ((Payload->flags &
-         static_cast<uint64_t>(xpti::payload_flag_t::NameAvailable))) {
-      // Add the kernel name/function name to the string table
-      FuncId = MStringTableRef.add(Payload->name, &Payload->name);
-    }
+    if (g_use_strict_hash) {
+      // If the payload's function name is available, add it to the string table
+      // and get its id
+      if ((Payload->flags &
+           static_cast<uint64_t>(xpti::payload_flag_t::NameAvailable))) {
+        // Add the kernel name/function name to the string table
+        FuncId = MStringTableRef.add(Payload->name, &Payload->name);
+      }
 
-    // If the payload's source file is available, add it to the string table
-    // and get its id Also, get the line number and column number from the
-    // payload
-    if ((Payload->flags &
-         static_cast<uint64_t>(xpti::payload_flag_t::SourceFileAvailable))) {
-      // Add source file information ot string table
-      FileId = MStringTableRef.add(Payload->source_file, &Payload->source_file);
-      LineNo = Payload->line_no;
-      ColNo = Payload->column_no;
+      // If the payload's source file is available, add it to the string table
+      // and get its id Also, get the line number and column number from the
```

Comment:
we are not getting column and line number any more here. is it expected?

Reply:
Done!